lib/presentation/widgets/my_image.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+class MyImage extends StatelessWidget {
+  final String? url;
+
+  const MyImage(this.url, {Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
We no need this Widget, in place we are gonna use the image we have to check:
if (url != null) {
  Image.network(url, fit: BoxFit.fitWidth); 
} else {
  // show placeholder
}

Reply:
done, but for what purpose check if parameter cannot be null

Reply:
Did not get you question. Based on your code this parameter can be null:
```
    if (url != null) {
      return Image.network(url!, fit: BoxFit.fitWidth);
    }
    return const Text('Image unavailable');
```

Reply:
ok